### PR TITLE
Add topic stubs to Synapse Docs

### PR DESCRIPTION
**Pull request by H-Shay**
@anoadragon453 and I were discussing how to improve Synapse's developer documentation and we came up with an outline of topics that we would like to see added to the docs. This PR adds those topic stubs to the table of contents, with the hope that the team and contributors alike will then help fill in the stubs and improve the documentation going forwards.

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -88,19 +88,158 @@
...
-    - [Log Contexts](log_contexts.md)
-    - [Replication](replication.md)
-    - [TCP Replication](tcp_replication.md)
-  - [Internal Documentation](development/internal_documentation/README.md)
+  - [Synapse Architecture](development/synapse_architechture.md)
+    - [Overview]()
+      - Flows of data through Synapse
+        - Starts with a Request/BG Job/Module
+        - Processing
+        - Maybe Response
```
Comment (anoadragon453) [inline docs/SUMMARY.md:96]: Others should note that the non-markdown links have no effect in `SUMMARY.md`, and are instead presented here as ideas for headings in pages / general notes.

They are left here to give one an idea of what a page may contain, and will be removed/transferred to doc pages before this PR merges.

**Review by anoadragon453 — COMMENTED** [review]
Thanks for turning this into a PR Shay!

I would be interested in other's opinions on the topics of the docs laid out here.

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -88,19 +88,158 @@
...
+      - Workers/Replication/Redis/Organization of a large homeserver instance
+  - [Configuration]()
+    - Experimental config options
+      - Define how long they should live after the corresponding msc has merged
+  - [Logging]()
+    - combine with [Log Contexts](log_contexts.md)
+  - [Storage]()
+    - What is current database schema?
+      - Start by linking script that can compile full schema
+      - Is there a tool that can visualize a postgres db? - Shay to check
```
Comment (MadLittleMods) [inline docs/SUMMARY.md:115]: I like this section 👍

Some copy-pastable snippets on how to explore the database would also be nice (cheat sheet):

>  - `psql synapse`: Open the postgres CLI against the `synapse` database
>  - `\q` -> `enter`: To quit/exit
>  - `\dt`: list tables
>  - `\d+ issues`: list columns and indexes for issues table
>  - `CREATE TABLE board_labels();`: Create a table called board_labels
>  - `DROP TABLE insertion_event_extremeties`
>  - `explain SELECT * FROM events WHERE depth = 1;`
>  - `explain analyze SELECT * FROM events WHERE depth = 1;`
>
> *-- https://github.com/matrix-org/synapse/pull/10245*

File: docs/SUMMARY.md [hunk 3]
```diff
@@ -88,19 +88,158 @@
...
+    - StreamReplication classes and clients
+    - How this interacts with sync?
+    - Stream token generators: Single Writer Only (sqlite), MultiWriterStreamIDGenerator (Postgres)
+  - [Notifier]()
+  - [Monitoring and Metrics]() # opentracing, phone home and metrics
+    - [Opentracing]()
+      - How does this work?
+      - How is this helpful?
+      - What's jaeger?
+      - How do I create and mutate a span?
```
Comment (MadLittleMods) [inline docs/SUMMARY.md:196]: 👍 Definitely interested in how to work with and use all of the jaeger stuff!

**Comment by clokep** [comment]
@H-Shay, @anoadragon453 Do you think we should land this or close it?

**Comment by H-Shay** [comment]
Fair question, I am interested in landing something like this, even if it doesn't go in the developer documentation (I think there were concerns with it becoming overwhelming?). Let's close this for now and maybe I'll flesh out something that could go in contrib or stand on it's own as an overview of Synapse.

**Pull request closed without merging**